### PR TITLE
EVEREST-629 Allowed pitr restore to new db cluster

### PR DESCRIPTION
**Pull request by filipmikes1**
EVEREST-629

### CODE REVIEW

File: apps/everest/src/modals/restore-db-modal/restore-db-modal.tsx [hunk 1]
```diff
@@ -83,14 +83,26 @@ const RestoreDbModal = <T extends FieldValues>({
...
+              },
+            });
+          } else {
+            navigate('/databases/new', {
+              state: {
+                selectedDbCluster: dbCluster.metadata.name,
+                backupName: pitrData!.latestBackupName,
+                namespace,
+                backupStorageName: selectedBackup,
+                pointInTimeDate: pitrBackup!.toISOString().split('.')[0] + 'Z',
```
Comment (fabio-silva) [inline apps/everest/src/modals/restore-db-modal/restore-db-modal.tsx:102]: Let's do some proper checking before using `!` and accessing `[0]`.
We can create an empty var at the beginning of the code block, do checks and eventually assign it to this expression.

```
let pointInTimeDate = '';

if (pitrBackup && pitrBackup instanceof Date ) {
    pointInTimeDate = pitrBackup.toISOString().split('.')[0] + 'Z';
}
```

**Review by fabio-silva — CHANGES_REQUESTED** [review]

File: apps/everest/src/modals/restore-db-modal/restore-db-modal.tsx (hunk 1 above)
Comment (filipmikes1) [reply apps/everest/src/modals/restore-db-modal/restore-db-modal.tsx:102]: Sure, I just copied it from few lines bellow. I can change it and refactor also old code to properly check it

File: apps/everest/src/hooks/api/db-cluster/useCreateDbCluster.ts [hunk 2]
```diff
@@ -123,6 +123,10 @@ const formValuesToPayloadMapping = (
       ...(backupDataSource?.dbClusterBackupName && {
         dataSource: {
           dbClusterBackupName: backupDataSource.dbClusterBackupName,
+          pitr: backupDataSource.pitr && {
```
Comment (fabio-silva) [inline apps/everest/src/hooks/api/db-cluster/useCreateDbCluster.ts:126]:
```
...(backupDataSource.pitr && {
    pitr: {
        date: backupDataSource.pitr.date,
        type: 'date',
    }
 })
```

**Review by fabio-silva — CHANGES_REQUESTED** [review]

**Review by solovevayaroslavna — COMMENTED** [review]
for me it looks okey, but... do we need a test there?

**Review by solovevayaroslavna — APPROVED** [review]

**Comment by filipmikes1** [comment]
> for me it looks okey, but... do we need a test there?

I will merge it. They can be created as a separated task.

**Pull request merged**